### PR TITLE
python310Packages.pillow: 9.4.0 -> 9.5.0

### PR DESCRIPTION
**Pull request by roblabla**
###### Description of changes

Pillow changelog: https://github.com/python-pillow/Pillow/blob/main/CHANGES.rst#950-2023-04-01

Note that this MR reverts an upstream commit (04cf5e2cfc5dc1676efd9f5c8d605ddfccb0f9ee) as it prevents pillow from finding its native dependencies during the build. We should probably try to fix the underlying issue, but I'm having trouble understanding what the patch even does.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/pillow/default.nix [hunk 1]
```diff
@@ -12,17 +12,21 @@
...
   disabled = pythonOlder "3.7";
 
   src = fetchPypi {
     pname = "Pillow";
     inherit version;
-    hash = "sha256-ocLXeARI65P7zDeJvzkWqlcg2ULjeUX0BWaAMX8c0j4=";
+    hash = "sha256-v1SEedM2cm16Ds6252fhefveN4M65CeUYCYxoHDWMPE=";
   };
 
+  # Reverts 04cf5e2cfc5dc1676efd9f5c8d605ddfccb0f9ee, which prevents pillow from
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/pillow/default.nix:26]: We can't use fetchpatch with the revert argument?

Comment (roblabla) [reply pkgs/development/python-modules/pillow/default.nix:26]: maybe, I didn't know fetchpatch had a revert argument!

**Comment by SuperSandro2000** [comment]
> The header issue has been fixed upstream in [python-pillow/Pillow#7069](https://github.com/python-pillow/Pillow/pull/7069).

Did you consider just fetching this commit instead?

---

This PR rebuilds a lot of packages which means we must target staging. Please follow the [contributing guide](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#rebasing-between-branches-ie-from-master-to-staging) to not potentially ping a lot of people.

**Review by asymmetric — CHANGES_REQUESTED** [review]
Agreed, this should just apply the patch from https://github.com/python-pillow/Pillow/pull/7069

**Comment by mweinelt** [comment]
As this seems stale I created #241913 to supersede it.

**Pull request closed without merging**